tools/common/Application.cs

```diff
@@ -490,19 +490,26 @@ public void RunRegistrar ()
 			ps.AssemblyResolver = resolver;
 			resolvedAssemblies.Add (ps.AssemblyResolver.Resolve (AssemblyNameReference.Parse ("mscorlib"), new ReaderParameters ()));
 
-			var rootName = Path.GetFileNameWithoutExtension (RootAssembly);
-			if (rootName != Driver.GetProductAssembly (this))
-				throw ErrorHelper.CreateError (66, "Invalid build registrar assembly: {0}", RootAssembly);
+			int index = 0;
```

Comment:
This could be a bool for first or even just move the check up before the foreach. The index is not used except for that check fwict

Reply:
👍